[Bug] Kotlin - Can't commit - error message not useful

Related: #9019

Neither of the error messages in the IDE indicate the problem, there's lots of noise which can likely be reduced, possibly just with a simple `gradlew ktLintFormat failed` in the header

```
git rev-parse HEAD
f6296af686e2eac4dee0ccb80a89ff92fd4e9751

(based off 765b8253c73c07addc84a373a2a5dd0d726166f1)
```

![image](https://user-images.githubusercontent.com/62114487/121529595-752cb700-c9f4-11eb-80ab-c13d70ce6fd3.png)


```
git commit -m "Add Analytics for Critical Exceptions"
Running Ktlint over these files:
AnkiDroid/src/main/java/com/ichi2/anki/analytics/CriticalException.kt
AnkiDroid/src/test/java/com/ichi2/anki/analytics/CriticalExceptionTest.kt
> Task :loadKtlintReporters UP-TO-DATE
> Task :lint-rules:loadKtlintReporters UP-TO-DATE
> Task :AnkiDroid:loadKtlintReporters UP-TO-DATE
> Task :api:loadKtlintReporters UP-TO-DATE
> Task :AnkiDroid:runKtlintFormatOverAmazonDebugSourceSet NO-SOURCE
> Task :api:runKtlintFormatOverAndroidTestDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAmazonDebugSourceSetFormat SKIPPED
> Task :lint-rules:runKtlintFormatOverKotlinScripts NO-SOURCE
> Task :AnkiDroid:runKtlintFormatOverAmazonReleaseSourceSet NO-SOURCE
> Task :lint-rules:ktlintKotlinScriptFormat SKIPPED
> Task :AnkiDroid:ktlintAmazonReleaseSourceSetFormat SKIPPED
> Task :api:ktlintAndroidTestDebugSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :api:ktlintAndroidTestReleaseSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverAndroidTestSourceSet NO-SOURCE
> Task :api:ktlintAndroidTestSourceSetFormat SKIPPED
> Task :lint-rules:runKtlintFormatOverMainSourceSet NO-SOURCE
> Task :AnkiDroid:runKtlintFormatOverAmazonSourceSet NO-SOURCE
> Task :lint-rules:ktlintMainSourceSetFormat SKIPPED
> Task :AnkiDroid:ktlintAmazonSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverDebugSourceSet NO-SOURCE
> Task :api:ktlintDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverAndroidTestAmazonDebugSourceSet NO-SOURCE
> Task :lint-rules:runKtlintFormatOverTestSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestAmazonDebugSourceSetFormat SKIPPED
> Task :lint-rules:ktlintTestSourceSetFormat SKIPPED
> Task :lint-rules:ktlintFormat UP-TO-DATE
> Task :AnkiDroid:runKtlintFormatOverAndroidTestAmazonSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestAmazonSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverAndroidTestDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverAndroidTestPlayDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestPlayDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverAndroidTestPlaySourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestPlaySourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverAndroidTestReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintAndroidTestReleaseSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverKotlinScripts NO-SOURCE
> Task :AnkiDroid:runKtlintFormatOverAndroidTestSourceSet NO-SOURCE
> Task :api:ktlintKotlinScriptFormat SKIPPED
> Task :AnkiDroid:ktlintAndroidTestSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintDebugSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverMainSourceSet NO-SOURCE
> Task :api:ktlintMainSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverKotlinScripts NO-SOURCE
> Task :api:runKtlintFormatOverReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintKotlinScriptFormat SKIPPED
> Task :runKtlintFormatOverKotlinScripts NO-SOURCE
> Task :ktlintKotlinScriptFormat SKIPPED
> Task :ktlintFormat UP-TO-DATE
> Task :api:ktlintReleaseSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverTestDebugSourceSet NO-SOURCE
> Task :api:ktlintTestDebugSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverTestReleaseSourceSet NO-SOURCE
> Task :api:ktlintTestReleaseSourceSetFormat SKIPPED
> Task :api:runKtlintFormatOverTestSourceSet NO-SOURCE
> Task :api:ktlintTestSourceSetFormat SKIPPED
> Task :api:ktlintFormat UP-TO-DATE
> Task :AnkiDroid:runKtlintFormatOverPlayDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintPlayDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverPlayReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintPlayReleaseSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverPlaySourceSet NO-SOURCE
> Task :AnkiDroid:ktlintPlaySourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintReleaseSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestAmazonDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestAmazonDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestAmazonReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestAmazonReleaseSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestAmazonSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestAmazonSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestPlayDebugSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestPlayDebugSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestPlayReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestPlayReleaseSourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestPlaySourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestPlaySourceSetFormat SKIPPED
> Task :AnkiDroid:runKtlintFormatOverTestReleaseSourceSet NO-SOURCE
> Task :AnkiDroid:ktlintTestReleaseSourceSetFormat SKIPPED

> Task :AnkiDroid:runKtlintFormatOverMainSourceSet
Execution optimizations have been disabled for task ':AnkiDroid:runKtlintFormatOverMainSourceSet' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:runKtlintFormatOverMainSourceSet' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an expl
icit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://do
cs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:runKtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or im
plicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.
org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\java'. Reason: Task ':AnkiDroid:run
KtlintFormatOverMainSourceSet' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency
. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/usergui
de/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\java'. Reason: Task ':AnkiDroid:run
KtlintFormatOverMainSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can
 lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validat
ion_problems.html#implicit_dependency for more details about this problem.

> Task :AnkiDroid:runKtlintFormatOverTestSourceSet
Execution optimizations have been disabled for task ':AnkiDroid:runKtlintFormatOverTestSourceSet' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:runKtlintFormatOverTestSourceSet' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an expl
icit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://do
cs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:runKtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or im
plicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.
org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\test\java'. Reason: Task ':AnkiDroid:run
KtlintFormatOverTestSourceSet' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency
. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/usergui
de/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\test\java'. Reason: Task ':AnkiDroid:run
KtlintFormatOverTestSourceSet' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit dependency. This can
 lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0.2/userguide/validat
ion_problems.html#implicit_dependency for more details about this problem.

> Task :AnkiDroid:ktlintMainSourceSetFormat
Execution optimizations have been disabled for task ':AnkiDroid:ktlintMainSourceSetFormat' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporterProviders
.bin'. Reason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an exp
licit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://d
ocs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporterProviders
.bin'. Reason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or i
mplicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle
.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an explicit or
 implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.grad
le.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit
dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0
.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\runKtlintFormatOv
erMainSourceSet_errors.bin'. Reason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' wi
thout declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Ple
ase refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\runKtlintFormatOv
erMainSourceSet_errors.bin'. Reason: Task ':AnkiDroid:ktlintMainSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without decl
aring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer
to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.

> Task :AnkiDroid:ktlintTestSourceSetFormat FAILED
Execution optimizations have been disabled for task ':AnkiDroid:ktlintTestSourceSetFormat' to ensure correctness due to the following reasons:
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporterProviders
.bin'. Reason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an exp
licit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://d
ocs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporterProviders
.bin'. Reason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or i
mplicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle
.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' without declaring an explicit or
 implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.grad
le.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\reporters.bin'. R
eason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without declaring an explicit or implicit
dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer to https://docs.gradle.org/7.0
.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\runKtlintFormatOv
erTestSourceSet_errors.bin'. Reason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':AnkiDroid:runKtlintFormatOverKotlinScripts' wi
thout declaring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Ple
ase refer to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
  - Gradle detected a problem with the following location: 'C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\intermediates\ktLint\runKtlintFormatOv
erTestSourceSet_errors.bin'. Reason: Task ':AnkiDroid:ktlintTestSourceSetFormat' uses this output of task ':runKtlintFormatOverKotlinScripts' without decl
aring an explicit or implicit dependency. This can lead to incorrect results being produced, depending on what order the tasks are executed. Please refer
to https://docs.gradle.org/7.0.2/userguide/validation_problems.html#implicit_dependency for more details about this problem.
C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\test\java\com\ichi2\anki\analytics\CriticalExceptionTest.kt:19:1 Wildcard import (cannot be auto-cor
rected)

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/7.0.2/userguide/command_line_interface.html#sec:command_line_warnings

Execution optimizations have been disabled for 4 invalid unit(s) of work during this build to ensure correctness.
Please consult deprecation warnings for more details.
8 actionable tasks: 4 executed, 4 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':AnkiDroid:ktlintTestSourceSetFormat'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.ConsoleReportWorkAction
   > KtLint found code style violations. Please see the following reports:
     - C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\build\reports\ktlint\ktlintTestSourceSetFormat\ktlintTestSourceSetFormat.txt

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```

--- a/pre-commit
+++ b/pre-commit
@@ -11,7 +11,10 @@ fi;
 echo "Running Ktlint over these files:"
 echo "$CHANGED_FILES"
 
-./gradlew ktlintFormat
+# -q removes noise from the output if it fails.
+# TODO: -w is better, but https://github.com/JLLeitschuh/ktlint-gradle/issues/457 adds noise
+# -w should display: "CriticalExceptionTest.kt:19:1 Wildcard import (cannot be auto-corrected)"
+./gradlew -q ktlintFormat
 
 echo "Completed ./gradlew ktlintFormat run."
 echo "$CHANGED_FILES" | while read -r file; do

